api/controllers/method.js

```diff
@@ -133,7 +121,123 @@ async function postMethodNewHttp(req, res) {
+  const newMethod = req.body;
+
+  // console.log(
+  //   "Received tools_techniques_types from client: >>> \n%s\n",
+  //   JSON.stringify(newMethod.tools_techniques_types, null, 2)
+  // );
+  // save any changes to the user-submitted text
+  const {
+    updatedText,
+    author,
+    oldArticle: oldMethod
+  } = await maybeUpdateUserText(req, res, "method", keyFieldsToObjects);
+  // console.log("updatedText: %s", JSON.stringify(updatedText));
+  // console.log("author: %s", JSON.stringify(author));
+  const [updatedMethod, er] = getUpdatedMethod(
+    user,
+    params,
+    newMethod,
+    oldMethod
+  );
+  // console.log("new method: \n%s", JSON.stringify(newMethod, null, 2));
+  // console.log("old method: \n%s", JSON.stringify(oldMethod, null, 2));
+  // console.log("updated method : \n%s", JSON.stringify(updatedMethod, null, 2));
+  if (!er.hasErrors()) {
+    if (updatedText) {
+      await db.tx("update-method", t => {
+        return t.batch([
+          t.none(INSERT_AUTHOR, author),
+          t.none(INSERT_LOCALIZED_TEXT, updatedText),
+          t.none(UPDATE_METHOD, updatedMethod)
+          // t.none("REFRESH MATERIALIZED VIEW search_index_en;")
+        ]);
+      });
+    } else {
+      await db.tx("update-method", t => {
+        return t.batch([
+          t.none(INSERT_AUTHOR, author),
+          t.none(UPDATE_METHOD, updatedMethod)
+          // t.none("REFRESH MATERIALIZED VIEW search_index_en;")
+        ]);
+      });
+    }
+    // the client expects this request to respond with json
+    // save successful response
+    // console.log("Params for returning method: %s", JSON.stringify(params));
+    const freshArticle = await getMethod(params);
+    // console.log("fresh article: %s", JSON.stringify(freshArticle, null, 2));
+    res.status(200).json({
+      OK: true,
+      article: freshArticle
+    });
+  } else {
+    console.error("Reporting errors: %s", er.errors);
+    res.status(400).json({
+      OK: false,
+      errors: er.errors
+    });
+  }
+}
+
+function getUpdatedMethod(user, params, newMethod, oldMethod) {
+  const updatedMethod = Object.assign({}, oldMethod);
+  const er = new ErrorReporter();
+  const cond = (key, fn) =>
+    setConditional(updatedMethod, newMethod, er, fn, key);
+  // admin-only
+  if (user.isadmin) {
+    cond("featured", as.boolean);
+    cond("hidden", as.boolean);
+    cond("original_language", as.text);
+    cond("post_date", as.date);
+  }
+
+  // media lists
+  ["links", "videos", "audio"].map(key => cond(key, as.media));
+  // photos and files are slightly different from other media as they have a source url too
+  ["photos", "files"].map(key => cond(key, as.sourcedMedia));
+  // boolean (would include "published" but we don't really support it)
+  ["facilitator"].map(key => cond(key, as.yesno));
+  // key
+  [
+    "facetoface_online_or_both",
+    "public_spectrum",
+    "open_limited",
+    "recruitment_method",
+    "level_polarization",
+    "scope_of_influence"
```

Comment:
`scope_of_influence` belongs in the array for `as.methodkeys`. looks like you have duplicated it here.

Reply:
Good catch, I thought I'd fixed that already. Important note: in cases `scope_of_influence` is a single key value, but in methods it is a list of key values.